alpaca/data/historical/news.py

```diff
@@ -54,12 +57,85 @@ def get_news(self, request_params: NewsRequest) -> Union[RawData, NewsSet]:
         """Returns news data
 
         Args:
-            request_params (NewsRequest): The request params to filter the news data"""
-        response = self.get(
-            path="/news",
-            data=request_params.to_request_fields(),
+            request_params (NewsRequest): The request params to filter the news data
+            feed (News)
+        Returns:
+            Union[RawData, NewsSet]: The news data
+        """
+
+        params = request_params.to_request_fields()
+        # paginated get request for news data api
+        raw_news = self._data_get(
+            symbol_or_symbols=request_params.symbols,
+            **params,
         )
+
         if self._use_raw_data:
-            return response
+            return raw_news
+
+        return NewsSet(raw_news)
+
+    # TODO: Remove duplication
+    def _data_get(
+        self,
+        symbol_or_symbols: Union[str, List[str]],
+        endpoint_asset_class: str = "news",
+        api_version: str = "v1beta1",
+        limit: Optional[int] = None,
```

Comment:
prefer to set small limit (may be 10?) as default. When I have tried this function, it takes longer time to get response.